addon/commands/insert-newLi-command.ts

```diff
@@ -18,301 +14,53 @@ export default class InsertNewLiCommand extends Command {
     super(model);
   }
 
-  canExecute(selection: ModelSelection = this.model.selection): boolean {
-    if(!selection.lastRange) {
+  canExecute(range: ModelRange | null = this.model.selection.lastRange): boolean {
+    if (!range) {
       return false;
     }
-    return selection.lastRange.hasCommonAncestorWhere(elementHasType("ul", "ol"));
+
+    return range.hasCommonAncestorWhere(ModelNodeUtils.isListContainer);
   }
+
   @logExecute
-  execute(): void {
-    const selection = this.model.selection;
-    const range = selection.lastRange;
+  execute(range: ModelRange | null = this.model.selection.lastRange): void {
     if (!range) {
-      throw new Error("couldn't get range");
+      throw new MisbehavedSelectionError();
     }
-    const startPosition = range.start;
-    const endPosition = range.end;
 
-    const isCollapsed = selection.isCollapsed;
-
-    const startParentLi = startPosition.findAncestors(node => ModelNode.isModelElement(node) && node.type === "li")[0];
-    const endParentLi = endPosition.findAncestors(node => ModelNode.isModelElement(node) && node.type === "li")[0];
+    const startParentLi = range.start.findAncestors(ModelNodeUtils.isListElement)[0];
+    const endParentLi = range.end.findAncestors(ModelNodeUtils.isListElement)[0];
 
     if (!startParentLi || !endParentLi) {
-      throw new Error("couldn't locate parent lis");
-    }
-
-    //collapsed selection case
-    if (isCollapsed) {
-      this.insertNewLi(startPosition);
-    }
-    //single li expanded selection case
-    else if (startParentLi === endParentLi) {
-      let newRange;
-      const text = new ModelText(INVISIBLE_SPACE);
-
-      this.model.change(mutator => {
-        mutator.insertNodes(range, text);
-        const cursorPos = ModelPosition.fromAfterNode(text);
-        newRange = new ModelRange(cursorPos, cursorPos);
-        this.model.selectRange(newRange);
-      });
-
-      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-      // @ts-ignore
-      this.insertNewLi(newRange.start);
-    }
-    //multiple lis selected case
-    else {
-      this.deleteRange(range);
+      throw new Error("Couldn't locate parent lis");
```

Comment:
Try to use custom errors as defined in errors.ts instead for more semantic errors
(also allows for error handling based on error type, which is the main reason for it.)